compiler/rustc_codegen_ssa/src/back/metadata.rs

```diff
@@ -379,6 +379,30 @@ pub(super) fn elf_e_flags(architecture: Architecture, sess: &Session) -> u32 {
             };
             e_flags
         }
+        Architecture::PowerPc64 => {
+            const EF_PPC64_ABI_UNKNOWN: u32 = 0;
+            const EF_PPC64_ABI_ELF_V1: u32 = 1;
+            const EF_PPC64_ABI_ELF_V2: u32 = 2;
+            
+            match sess.target.options.abi.as_ref() {
+                // If the flags do not correctly indicate the ABI,
+                // linkers such as ld.lld assume that the ppc64 object files are always ELFv2
+                // which leads to broken binaries if ELFv1 is used for the object files.
+                "elfv1" => EF_PPC64_ABI_ELF_V1,
+                "elfv2" => EF_PPC64_ABI_ELF_V2,
+                // If options.abi is empty, then try to figure out which ABI to use.
+                // Based loosely on llvm check (check llvm/include/llvm/TargetParser/Triple.h#isPPC64ELFv2ABI and clang/lib/Basic/Targets/PPC.h#PPC64TargetInfo)
+                "" => {
+                    if sess.target.llvm_target.contains("bsd") || sess.target.llvm_target.contains("musl") {
+                        EF_PPC64_ABI_ELF_V2
+                    } else {
+                        EF_PPC64_ABI_ELF_V1
+                    }
+                },
+                // Fall back
+                _ => EF_PPC64_ABI_UNKNOWN,
+            }
```

Comment:
We should not assume. We should panic, and submit a separate PR proposing to add the elf version to the `abi` field for all PowerPC targets that use an ELF format.

Reply:
Done, but why should the ABI be specified in a seperate PR? Now all predefined ppc64 targets are basically broken?